Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift

```diff
@@ -0,0 +1,336 @@
+        assertOptionIsIncluded(.threadInfo) { message, _ in
+            message.deletedAt == nil && (message.isRootOfThread || message.isPartOfThread)
+        }
+    }
+
+    func test_reactions() {
+        for reactionsEnabled in [true, false] {
+            // Create channel with the given `reactionsEnabled` config
+            let channel: ChatChannel = .mock(
+                cid: .unique,
+                config: .mock(
+                    reactionsEnabled: reactionsEnabled
+                )
+            )
+
+            // Iterate channel history and check `.reactions` is there if conditions are met
+            assertOptionIsIncluded(.reactions, channel: channel) { message, _ in
+                message.deletedAt == nil && !message.reactionScores.isEmpty && reactionsEnabled
+            }
+        }
+    }
+
+    func test_errorIndicator() {
+        // Iterate channel history and check `.errorIndicator` is there if conditions are met
+        assertOptionIsIncluded(.errorIndicator) { message, _ in
+            message.deletedAt == nil && message.lastActionFailed
+        }
+    }
+    
+    func test_isMessageStandaloneOrLastInSequence() {
+        // Iterate all test cases
+        for messageIndex in 0..<channelHistory.count {
+            let (_, isStandaloneOrLastInSequence) = channelHistory[messageIndex]
+
+            let isStandaloneOrLastInSequenceActual = optionsResolver.isMessageStandaloneOrLastInSequence(
+                messageIndexPath: .init(item: messageIndex, section: 0),
+                messages: .init(channelHistory.map(\.0))
+            )
+
+            // Assert expected and actual values match.
+            XCTAssertEqual(
+                isStandaloneOrLastInSequenceActual,
+                isStandaloneOrLastInSequence
+            )
+        }
+    }
+
+    // MARK: - Helpers
+
+    private func assertOptionIsIncluded(
+        _ option: ChatMessageLayoutOptions,
+        channel: ChatChannel = .mock(cid: .unique),
+        mustBeIncludedFor: (_ message: ChatMessage, _ isStandaloneOrLastInSequence: Bool) -> Bool
+    ) {
+        // Get the list of messages from channel history
+        let messages = AnyRandomAccessCollection(channelHistory.map(\.0))
+
+        // Iterate channel history
+        for messageIndex in 0..<channelHistory.count {
+            // Get message data
+            let (message, isStandaloneOrLastInSequence) = channelHistory[messageIndex]
+
+            // Calculate layout options for the current message
+            let layoutOptions = optionsResolver.optionsForMessage(
+                at: .init(item: messageIndex, section: 0),
+                in: channel,
+                with: messages
+            )
+
+            // Assert the `option` is there or not depending on the requirement
+            XCTAssertEqual(
+                layoutOptions.contains(option),
+                mustBeIncludedFor(message, isStandaloneOrLastInSequence)
+            )
+        }
+    }
+}
```

Comment:
Like I said above, with this helper, is not possible to check the exact message which added the option, which will not catch possible future bugs

Reply:
✅ Removed